compiler/p/codegen/OMRMemoryReference.cpp

```diff
@@ -1609,106 +1652,76 @@ void OMR::Power::MemoryReference::accessStaticItem(TR::Node *node, TR::SymbolRef
+      TR_ExternalRelocationTargetKind relocationKind;
+      intptrj_t                        addr;
 
-         if (symbol->isConst())
-            {
-            relocationTarget = (uint8_t *)ref->getOwningMethod(comp)->constantPool();
-            relocationKind   = TR_ConstantPoolOrderedPair;
-            }
-         else if (symbol->isClassObject())
-            {
-            relocationKind   = TR_ClassAddress;
-            }
-         else if (symbol->isMethod())
-            {
-            relocationTarget = (uint8_t *)ref;
-            relocationKind   = TR_MethodObject;
-            }
-         else
-            {
-            relocationTarget = (uint8_t *)ref;
-            relocationKind   = TR_DataAddress;
-            }
+      addr = symbol->isStatic() ? (intptrj_t)symbol->getStaticSymbol()->getStaticAddress() :
+                                  (intptrj_t)symbol->getMethodSymbol()->getMethodAddress();
 
-         if (comp->getOption(TR_AOT))
-            {
-             TR_RelocationRecordInformation *recordInfo = ( TR_RelocationRecordInformation *)comp->trMemory()->allocateMemory(sizeof( TR_RelocationRecordInformation), heapAlloc);
-            recordInfo->data1 = (uintptr_t)relocationTarget;
-            recordInfo->data2 = node ? (uintptr_t)node->getInlinedSiteIndex() : (uintptr_t)-1;
-            recordInfo->data3 = orderedPairSequence1;
-            staticRelocation = new (cg->trHeapMemory()) TR::PPCPairedRelocation(rel1, rel2, (uint8_t *)recordInfo, relocationKind, node);
-            cg->getAheadOfTimeCompile()->getRelocationList().push_front(staticRelocation);
-
-            if (!specialCase)
-               self()->setStaticRelocation(staticRelocation);
-            }
+      if (!node) node = cg->getCurrentEvaluationTreeTop()->getNode();
+
+      if (symbol->isStartPC())
+         {
+         loadAddressConstantInSnippet(cg, node, 0, reg, NULL, TR::InstOpCode::addi, false, NULL);
+         return;
+         }
+
+      self()->setBaseModifiable();
+      rel1 = generateTrg1ImmInstruction(cg, TR::InstOpCode::lis, node, reg, cg->hiValue(addr)&0x0000ffff);
+      self()->addToOffset(node, LO_VALUE(addr), cg);
+
+      if (comp->compileRelocatableCode())
+         {
+         TR_ASSERT_FATAL(
+            symbol->isClassObject(),
+            "expected a class symbol for symref #%d\n",
+            ref->getReferenceNumber());
+
+         TR_RelocationRecordInformation *recordInfo = ( TR_RelocationRecordInformation *)comp->trMemory()->allocateMemory(sizeof( TR_RelocationRecordInformation), heapAlloc);
+         recordInfo->data1 = (uintptr_t)ref;
+         recordInfo->data2 = node ? (uintptr_t)node->getInlinedSiteIndex() : (uintptr_t)-1;
+         recordInfo->data3 = orderedPairSequence1;
+         staticRelocation = new (cg->trHeapMemory()) TR::PPCPairedRelocation(rel1, NULL, (uint8_t *)recordInfo, TR_ClassAddress, node);
+         cg->getAheadOfTimeCompile()->getRelocationList().push_front(staticRelocation);
```

Comment:
is there a reason you didn't want to use addAOTRelocation() here ?

Reply:
This is pushing onto `getRelocationList()`, not `getAOTRelocationList()`. The relocation will be picked up later from the former list, and its `mapRelocation()` will translate it into an `ExternalRelocation`, which will then be passed to `addAOTRelocation()`.

I know these lists' names aren't very evocative, but I "merely" didn't rename them.